actions/templates/install/index.js

```diff
@@ -0,0 +1,236 @@
+    const requiredHeaders = ['Authorization'];
+    let errorMessages = checkMissingRequestInputs(params, [], requiredHeaders);
+    if (errorMessages) {
+      return errorResponse(401, errorMessages, logger);
+    }
+    const requiredParams = [
+      POST_PARAM_NAME
+    ];
+    errorMessages = checkMissingRequestInputs(params, requiredParams);
+    if (errorMessages) {
+      return errorResponse(400, errorMessages, logger);
+    }
+
+    // extract the user Bearer token from the Authorization header
+    const accessToken = getBearerToken(params);
+
+    try {
+      // validate the token, an exception will be thrown for a non-valid token
+      await validateAccessToken(accessToken, imsUrl, imsClientId);
+    } catch (error) {
+      return errorResponse(401, [errorMessage(ERR_RC_INVALID_IMS_ACCESS_TOKEN, error.message)], logger);
+    }
+
+    Enforcer.v3_0.Schema.defineDataTypeFormat('string', 'uuid', null);
+    Enforcer.v3_0.Schema.defineDataTypeFormat('string', 'uri', null);
+
+    // WPAR002 - skip a warning about the "allowEmptyValue" property
+    // see https://swagger.io/docs/specification/describing-parameters/ Empty-Valued and Nullable Parameters
+    // EDEV001 - skip a warning about the basepath property, needed by IO Runtime for deploying apis
+    const openapi = await Enforcer('./template-registry-api.json', { componentOptions: { exceptionSkipCodes: ['WPAR002', 'EDEV001'] } });
+
+    const body = serializeRequestBody(params);
+
+    logger.debug(`Request body: ${JSON.stringify(body)}`);
+
+    const [req, reqError] = openapi.request({
+      method: HTTP_METHOD,
+      path: '/templates/{templateId}/install',
+      body
+    });
+    if (reqError) {
+      return errorResponse(400, [errorMessage(ERR_RC_INCORRECT_REQUEST, reqError.toString().split('\n').map(line => line.trim()).join(' => '))], logger);
+    }
+
+    const template = await findTemplateById(dbParams, params.templateId);
+    if (!template) {
+      return errorResponse(404, [errorMessage(ERR_RC_INVALID_TEMPLATE_ID, `Template with id ${params.templateId} not found.`)], logger);
+    }
+    logger.debug(`Template found: ${JSON.stringify(template)}`);
+
+    // extract credentials and APIs from the template
+    const { credentials, apis } = template;
+
+    logger.debug(`Credentials found: ${JSON.stringify(credentials)}`);
+    logger.debug(`APIs found: ${JSON.stringify(apis)}`);
+
+    // credential types could be: 'oauth_server_to_server', 'apikey', 'oauthnativeapp', 'oauthwebapp', 'oauthsinglepageapp'
+    // credential flow types could be: 'adobeid', 'entp'
+    // adobeid is used for OAuth creds and entp is used for OAuthS2S and JWT creds
+
+    const mapAdobeIdCredentialTypeToPlatformType = {
+      apikey: 'apiKey',
+      oauthnativeapp: 'NativeApp',
+      oauthwebapp: 'WebApp',
+      oauthsinglepageapp: 'SinglePageApp'
+    };
+
+    const consoleClient = await consoleLib.init(accessToken, params.IMS_CLIENT_ID, 'stage');
```

Comment:
users can install templates in either console stage or prod, so I think we need a way to set this based on the service's environment

Reply:
Yes good point, I was also thinking the same. I'll make the change required.

Reply:
I'm thinking about how to get the service's environment. Can we set `NODE_ENV` and then extract it by `process.env.NODE_ENV` or is there another better way?

Reply:
was originally thinking we could just introduce a new ENV param to all the actions in app.config.yaml and .env and set it to `stage` or `prod`

but maybe it'd be easier to just parse `process.env.__OW_API_HOST` to see if we're running in runtime stage / prod, then set our IMS env based on that? havent done this before but might be worth looking into https://developer.adobe.com/runtime/docs/guides/reference/environment_variables/

Reply:
Yeah sure, we can try this approach

Reply:
did we try this? still seeing process.env.NODE_ENV being used, which im not sure will work

Reply:
Yes sorry, I haven't yet changed it to use `process.env.__OW_API_HOST`. 
I was wondering whether this would work or not. I'm not sure but I guess namespaces always get created in runtime prod and we just use different namespaces for the service based on whether it is stage/prod. So I guess `__OW_API_HOST` would return runtime prod in both cases. I maybe wrong here so need your thoughts on this.

Reply:
Just tried getting process.env.__OW_API_HOST for my actions deployed in my namespace:
`apiHost: https://controller-gw-ns-team-ethos651prodjpn3-runtime-prod-b.ethos651-prod-jpn3.ethos.adobe.net`
Looks like it gives the whole ethos container name back.

Reply:
> So I guess __OW_API_HOST would return runtime prod in both cases

For namespaces made in dev console stage or prod, yes, however we've made the stage namespace for this service in runtime stage via the wskadmin cli, so that one should return a stage ethos container. Let's test this in a runtime stage namespace too and see if it works

Reply:
Okay sounds good. I'll test with the runtime stage namespace, hopefully, that should return stage container info.